homeassistant/components/tesla/binary_sensor.py

```diff
@@ -29,22 +28,17 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
 class TeslaBinarySensor(TeslaDevice, BinarySensorEntity):
     """Implement an Tesla binary sensor for parking and charger."""
 
-    def __init__(self, tesla_device, controller, sensor_type, config_entry):
+    def __init__(self, tesla_device, controller, config_entry):
         """Initialise of a Tesla binary sensor."""
         super().__init__(tesla_device, controller, config_entry)
-        self._state = False
-        self._sensor_type = sensor_type
+        self._state = None
+        self._sensor_type = tesla_device.sensor_type
```

Comment:
Please use the device class constants imported from the binary sensor integration.

Reply:
They're used in the underlying library.

Reply:
We shouldn't rely on the library directly for this as this is the home assistant api. There should be a link in this module between the library output and the home assistant defined device classes, eg a dictionary.

Reply:
Right now thats probably a map of 
```
BINARY_SENSOR_TYPES = {
    "connectivity": DEVICE_CLASS_CONNECTIVITY,
    "power": DEVICE_CLASS_POWER  
}
```
Ideally "connectivity" and "power" are constants imported from the pypi package.

If we ever change the underlying constant values then everything just works.

and then device_class looks like

`BINARY_SENSOR_TYPES.get(tesla_device.sensor_type)`

Reply:
It's not a difficulty question. It's the fact it's a cosmetic change that adds maintenance burdens since the underlying library is only used by HA, is so tightly coupled with HA there's an [outstanding issue](https://github.com/zabuldon/teslajsonpy/issues/24) to try to decouple it so it can be used by other programs, and specifically has a function called [`get_homeassistant_components`](https://github.com/zabuldon/teslajsonpy/blob/dev/teslajsonpy/controller.py#L480) to return HA compatible objects. 

We're proposing to literally define an identity map where the underlying component requires setting the [parameter](https://github.com/zabuldon/teslajsonpy/blob/dev/teslajsonpy/binary_sensor.py#L41) specifically to the hass `device_class`.

So literally the process for adding a sensor is to read the HA documentation, add it into the underlying library, and then create a useless identity map in HA. That's just dumb.

I'm maintaining the library, but I'm not volunteering to rewrite the library. If someone from HA wants to, they can and then we can finally decouple it. Then the identity map has some actual value.

Reply:
We don't need to rewrite the library to add this map here in the platform.

Reply:
Another solution that may have less maintenance is to check that the device class is in the defined list of device classes before returning it in the property.

https://github.com/home-assistant/core/blob/0b9663a23b763b2a44eb576f53ae990c48166314/homeassistant/components/binary_sensor/__init__.py#L97-L122

Reply:
The check within `DEVICE_CLASSES` seems useful. I'll implement that.